src/main/scala/com/imaginea/Main.scala

```diff
@@ -1715,7 +1807,66 @@ object Main extends App {
         }
       case _ => false
     }
+  }
+
+  def populateFilteredInstances(siteId: Long,  filters: List[Filter]) : Option[Site1] = {
+    val mayBeSite = cachedSite.get(siteId)
+    mayBeSite match {
+      case Some (site) =>
+        val instances = site.instances
+        val (siteFiltersToSave, filteredInstances) = if (filters.nonEmpty) {
+          val siteFilters = site.filters
+          val listOfSiteFilters = siteFilters.map {
+            siteFilter => SiteFilter (siteFilter.id, siteFilter.accountInfo, filters)
+          }
+          val listOfInstances = instances.flatMap {
+            instance =>
+              val filterExists = filters.find (filter => filterExistsInTags (instance, filter) )
+              filterExists.map (filter => instance)
+          }
+          (listOfSiteFilters, listOfInstances)
+        }
+        else {
+          (site.filters, instances)
+        }
+        val instancesToSave = getInstancesToSave (filteredInstances.toSet)
+        val lbsToSave = getLoadBalancersToSave (site.loadBalancers)
+        val sgsToSave = getScalingGroupsToSave (site.scalingGroups)
+        val rInstancesToSave = getReservedInstancesToSave (site.reservedInstanceDetails)
+        val siteToSave = Site1 (site.id, site.siteName, instancesToSave, rInstancesToSave, siteFiltersToSave, lbsToSave, sgsToSave, site.groupsList)
+        siteToSave.toNeo4jGraph (siteToSave)
+        cachedSite.put (siteId, siteToSave)
+        Some (siteToSave)
+      case None => logger.warn (s"could not get Site from cache for siteId : $siteId")
+        None
+    }
+  }
 
+  def populateInstances(site: Site1): Site1 = {
+    val siteFilters = site.filters
+    logger.info(s"Parsing instance : ${site.instances}")
+    val computedResult = siteFilters.foldLeft((List[Instance](), List[ReservedInstanceDetails](), List.empty[LoadBalancer], List.empty[ScalingGroup])) {
+      (result, siteFilter) =>
+        val accountInfo = siteFilter.accountInfo
+        val amazonEC2 = AWSComputeAPI.getComputeAPI(accountInfo)
+        val instances = AWSComputeAPI.getInstances(amazonEC2, accountInfo)
+        val reservedInstanceDetails = AWSComputeAPI.getReservedInstances(amazonEC2)
+        val loadBalancers = AWSComputeAPI.getLoadBalancers(accountInfo)
+        val scalingGroup = AWSComputeAPI.getAutoScalingGroups(accountInfo)
+        (result._1 ::: instances, result._2 ::: reservedInstanceDetails, result._3 ::: loadBalancers, result._4 ::: scalingGroup)
+    }
+    val site1 = Site1(None, site.siteName, computedResult._1, computedResult._2, site.filters, computedResult._3, computedResult._4, List())
+    site1.toNeo4jGraph(site1)
+    site1
+  }
 
+  def saveCachedSite(siteId: Long): Boolean = {
+    val mayBeSite = cachedSite.get(siteId)
+    mayBeSite.exists { site =>
```

Comment:
exists is shorter here. But that's not the purpose of exists.
Replace it with a foreach and mayBeSite.isDefined